tamr_unify_client/models/dataset/resource.py

```diff
@@ -178,6 +178,21 @@ def from_geo_features(self, features, geo_attr=None):
             self._features_to_updates(features, record_id, key_attrs, geo_attr)
         )
 
+    def upstream_datasets(self):
+        """The Dataset's upstream datasets.
+
+        :return: A list of the Dataset's upstream datasets.
+        """
+        from tamr_unify_client.models.dataset.upstreamDataset import UpstreamDataset
+
+        upstream_list = []
+        alias = self.api_path + "/upstreamDatasets"
+        resources = self.client.get(alias).successful().json()
+
+        for dataset_id in resources:
+            upstream_list.append(UpstreamDataset(self.client, dataset_id).dataset())
```

Comment:
Possibly remove `.dataset()` from the end so you're returning a list of `UpstreamDataset`s rather than `Dataset`s?